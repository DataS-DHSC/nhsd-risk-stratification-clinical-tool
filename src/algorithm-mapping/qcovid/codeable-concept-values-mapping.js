export const SNOMED_CODE_NONE = '260413007';
export const QCOVID_CODE_NONE = 'NONE';

/**
 * This file includes information and util functions needed to convert form fields
 * values for observations of type CODEABLE_CONCEPT
 */

export const VALUE_OBSERVATIONS = {
  // ethnicity
  bangladeshi: { label: 'Bangladeshi', snomedCode: '186003008' },
  chinese: { label: 'Chinese', snomedCode: '33897005' },
  indian: { label: 'Indian', snomedCode: '110751000000108' },
  pakistani: { label: 'Pakistani', snomedCode: '92461000000105' },
  'other-asian': {
    label: 'Another Asian background',
    snomedCode: '92701000000102',
  },
  african: { label: 'African', snomedCode: '92491000000104' },
  caribbean: { label: 'Caribbean', snomedCode: '107691000000105' },
  'other-black': {
    label: 'Another Black background',
    snomedCode: '92501000000105',
  },
  'asian-white': { label: 'Asian and White', snomedCode: '92441000000109' },
  'african-white': {
    label: 'Black African and White',
    snomedCode: '92431000000100',
  },
  'caribbean-white': {
    label: 'Black Caribbean and White',
    snomedCode: '92421000000102',
  },
  'other-mixed': {
    label: 'Another Mixed background',
    snomedCode: '92451000000107',
  },
  british: {
    label: 'British, English, Northern Irish, Scottish, or Welsh',
    snomedCode: '494131000000105',
  },
  irish: { label: 'Irish', snomedCode: '494161000000100' },
  'other-white': {
    label: 'Another White background',
    snomedCode: '92411000000108',
  },
  'other-ethnic-group': {
    label: 'Another ethnic background',
    snomedCode: '94151000000105',
  },
  'not-saying': { label: 'Prefer not to say', snomedCode: '92531000000104' },

  // sex
  female: { label: 'Female', snomedCode: '248152002' },
  male: { label: 'Male', snomedCode: '248153007' },

  // residential
  'not-homeless-or-resident-of-care-home': {
    label: 'Not homeless or resident of a care home',
    snomedCode: '160724009',
  },
  homeless: { label: 'Homeless', snomedCode: '266935003' },
  'resident-of-a-care-home': {
    label: 'Resident of a care home',
    snomedCode: '248171000000108',
  },

  // diabetes
  'has-no-diabetes': { label: 'Has no diabetes', snomedCode: QCOVID_CODE_NONE },
  'has-type-1-diabetes': { label: 'Has type 1', snomedCode: 'TYPE_1_DIABETES' },
  'has-type-2-diabetes': { label: 'Has type 2', snomedCode: 'TYPE_2_DIABETES' },

  // kidney, renal
  'no-cdk-or-has-ckd-1-or-2': {
    label: 'No CKD, or has CKD 1 or 2',
    snomedCode: QCOVID_CODE_NONE,
  },
  'has-cdk3': { label: 'Has CKD3', snomedCode: 'CKD3' },
  'has-cdk4': { label: 'Has CKD4', snomedCode: 'CKD4' },
  'has-cdk5-without-dialysis-or-transplant': {
    label: 'Has CKD5 without dialysis or transplant',
    snomedCode: 'CKD5_NO_DIALYSIS',
  },
  'has-cdk5-with-dialysis-in-the-last-12-months': {
    label: 'Has CKD5 with dialysis in the last 12 months',
    snomedCode: 'CKD5_DIALYSIS',
  },
  'has-cdk5-with-transplant': {
    label: 'Has CKD5 with transplant',
    snomedCode: 'CKD_TRANSPLANT',
  },

  // learning disability
  'has-no-learning-disability-or-downs-syndrome': {
    label: "Has no learning disability or Down's syndrome",
    snomedCode: QCOVID_CODE_NONE,
  },
  'has-learning-disability-excluding-downs-syndrome': {
    label: "Has learning disability excluding Down's syndrome",
    snomedCode: 'LEARNING_DISABILITY',
  },
  'has-downs-syndrome': {
    label: "Has Down's syndrome",
    snomedCode: 'DOWNS_SYNDROME',
  },

  // chemotherapy
  'no-chemotherapy-in-the-last-12-months': {
    label: 'No chemotherapy in the last 12 months',
    snomedCode: QCOVID_CODE_NONE,
  },

  // chemotherapy: group A
  '5fu-single-agent': {
    label: '5FU single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  abiraterone: { label: 'Abiraterone', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  anagrelide: { label: 'Anagrelide', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'aromatase-inhibitors': {
    label: 'Aromatase inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'bevacizumab-single-agent': {
    label: 'Bevacizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  bisphosphonate: {
    label: 'Bisphosphonate',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  busulfan: { label: 'Busulfan ', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'capecitabine-single-agent': {
    label: 'Capecitabine single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'cdk4-6-inhibitors': {
    label: 'CDK4/6 inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  cetuximab: { label: 'Cetuximab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'cisplatin-based-regimens': {
    label: 'Cisplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  denosumab: { label: 'Denosumab ', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  durvalumab: { label: 'Durvalumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  enzalutamide: { label: 'Enzalutamide', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  fulvestrant: { label: 'Fulvestrant', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  hydroxycarbamide: {
    label: 'Hydroxycarbamide',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'interferon-all-formulations': {
    label: 'Interferon (all formulations)',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  ipilimumab: { label: 'Ipilimumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  lenvatinib: { label: 'Lenvatinib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  methotrexate: { label: 'Methotrexate', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'mitomycin-c': { label: 'Mitomycin C', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'mtor-inhibitors': {
    label: 'mTOR inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  nivolumab: {
    label: 'Nivolumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  panitumumab: { label: 'Panitumumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'parp-inhibitors': {
    label: 'PARP inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  pembrolizumab: {
    label: 'Pembrolizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  pemetrexed: { label: 'Pemetrexed', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  raltitrexed: { label: 'Raltitrexed', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  regorafinib: { label: 'Regorafinib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'single-agent-atezolizumab': {
    label: 'Atezolizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  sorafenib: { label: 'Sorafenib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  tamoxifen: { label: 'Tamoxifen', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'taxane-weekly': {
    label: 'Taxane â€“ weekly',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'trastuzumab-pertuzumab': {
    label: 'Trastuzumab +/- pertuzumab',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'tyrosine-kinase-inhibitors-including-alk': {
    label: 'Tyrosine kinase inhibitors (including ALK &/or ROS)',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },

  // chemotherapy: group B
  abvd: { label: 'ABVD', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'anthracycline-based-regimens': {
    label: 'Anthracycline based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'asparaginase-based-regimens': {
    label: 'Asparaginase based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  beacopp: { label: 'BEACOPP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  bendamustine: { label: 'Bendamustine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'brentuximab-vedotin': {
    label: 'Brentuximab vedotin',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'btk-inhibitors': {
    label: 'BTK inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  cabazitaxel: { label: 'Cabazitaxel', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'carboplatin-based-regimens': {
    label: 'Carboplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  chop: { label: 'CHOP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  chorambucil: { label: 'Chorambucil', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cladrabine: { label: 'Cladrabine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cmf: { label: 'CMF', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cvad: { label: 'CVAD', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'cyclophosphamide-fludarabine-combinations': {
    label: 'Cyclophosphamide/Fludarabine combinations',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'dacarbazine-based-regimens': {
    label: 'Dacarbazine based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  daratumumab: { label: 'Daratumumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  dhap: { label: 'DHAP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  eshap: { label: 'ESHAP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'etoposide-based-regimens': {
    label: 'Etoposide based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  fec: { label: 'FEC', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  gdp: { label: 'GDP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  gemcitabine: { label: 'Gemcitabine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  ice: { label: 'ICE', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'ifophosphamide-based-regimens': {
    label: 'Ifophosphamide based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  imids: { label: 'IMIDs', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'irinotecan-and-oxaliplatin-based-regimens': {
    label: 'Irinotecan and Oxaliplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'jak-inhibitors': {
    label: 'JAK inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'liposomal-doxorubicin': {
    label: 'Liposomal doxorubicin',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  lomustine: { label: 'Lomustine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  mogalizumab: { label: 'Mogalizumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  mvac: { label: 'MVAC', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'nab-paclitaxel': {
    label: 'Nab-paclitaxel',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  obinutuzumab: { label: 'Obinutuzumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  pentostatin: { label: 'Pentostatin', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'pi3kinase-inhibitors': {
    label: 'PI3Kinase inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'proteosome-inhibitors': {
    label: 'Proteosome inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  rituximab: { label: 'Rituximab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'taxane-3-weekly': {
    label: 'Taxane â€“ 3 weekly',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  temozolomide: { label: 'Temozolomide', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  topotecan: { label: 'Topotecan', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'trastuzumab-emtansine': {
    label: 'Trastuzumab-emtansine',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  ventoclax: { label: 'Ventoclax', snomedCode: 'SACT_CHEMOTHERAPY_B' },

  // chemotherapy: group C
  'all-aml-regimens': {
    label: 'All ALL / AML regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
  bep: { label: 'BEP', snomedCode: 'SACT_CHEMOTHERAPY_C' },
  'highly-immunosuppressive-chemotherapy': {
    label:
      'Highly immunosuppressive chemotherapy (e.g. FluDAP, high dose Methotrexate & Cytarabine)',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
  'trifuradine-tipracil': {
    label: 'Trifuradine/Tipracil',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
};

export const getSnomedCodeForValue = (valueId) =>
  VALUE_OBSERVATIONS[valueId].snomedCode;

export const getLabelForSnomedCode = (snomedCode) => {
  if (snomedCode === SNOMED_CODE_NONE || snomedCode === QCOVID_CODE_NONE) {
    return 'None';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_A') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol A';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_B') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol B';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_C') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol C';
  }

  try {
    const { label } = Object.values(VALUE_OBSERVATIONS).find(
      (value) => value.snomedCode === snomedCode
    );
    return label;
  } catch (e) {
    throw new Error(`cannot find label for: ${snomedCode} : ${e.message}`);
  }
};

const getLabelForValue = (valueId) => VALUE_OBSERVATIONS[valueId].label;

export const getObservationOption = (name) => ({
  label: getLabelForValue(name),
  name,
});
