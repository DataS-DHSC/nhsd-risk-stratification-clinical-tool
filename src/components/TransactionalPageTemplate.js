import React from 'react';
import PropTypes from 'prop-types';
import PageTitle from './PageTitle';
import FormGenerator from './FormGenerator';
import PageReviewDates from './PageReviewDates';
import HeadTags from './HeadTags';
import FormIntroGenerator from './FormIntroGenerator';

const TransactionalPageTemplate = ({
  form,
  prevData,
  context,
  reviewDatesSource,
  featureFlags,
  formBaseUrl,
  pageNames,
}) => {
  const { csrfToken } = prevData;
  const [contextData, setField] = context();
  return (
    <div className="nhsuk-grid-row">
      <div className="nhsuk-grid-column-two-thirds">
        <HeadTags
          title={form.label}
          description={form.label}
          featureFlags={featureFlags}
        />

        <PageTitle label={form.label} />

        <FormIntroGenerator formId={form.formId} />

        <FormGenerator
          prevData={contextData ? { ...contextData, csrfToken } : prevData}
          form={form}
          setField={setField}
          formBaseUrl={formBaseUrl}
          pageNames={pageNames}
        />

        <PageReviewDates page={reviewDatesSource} />
      </div>
    </div>
  );
};

TransactionalPageTemplate.propTypes = {
  form: PropTypes.shape().isRequired,
  prevData: PropTypes.shape().isRequired,
  context: PropTypes.func.isRequired,
  reviewDatesSource: PropTypes.string.isRequired,
  formBaseUrl: PropTypes.string.isRequired,
  featureFlags: PropTypes.shape(),
  pageNames: PropTypes.arrayOf(PropTypes.string).isRequired,
};

TransactionalPageTemplate.defaultProps = {
  featureFlags: {},
};

export default TransactionalPageTemplate;
