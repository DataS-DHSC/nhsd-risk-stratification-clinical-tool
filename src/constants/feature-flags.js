export const ADOBE_ANALYTICS_FEATURE_FLAG = true;
