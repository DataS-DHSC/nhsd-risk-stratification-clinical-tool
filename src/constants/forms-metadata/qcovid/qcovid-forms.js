import { IMMU_HAEMA_CONDITIONS_FORM } from './immune-and-haematological-conditions';
import { BODY_MASS_INDEX_FORM } from './body-mass-index';
import { CARDIOVASCULAR_DISEASES_FORM } from './cardiovascular-diseases';
import { IMMUNO_CANCER_CONDITIONS_FORM } from './immunosuppressants-cancer-conditions';
import { METABOLIC_RENAL_FORM } from './metabolic-renal-liver-conditions';
import { NEURO_PSYCH_CONDITIONS_FORM } from './neurological-and-psychiatric-conditions';
import { PATIENT_DETAILS_FORM } from './patient-details';
import { RESP_DISEASES_TREATMENTS_FORM } from './respiratory-diseases-and-treatments';
import { extractDefaults } from '../../../utils/extractDefaults';
import { fromEntries } from '../../../utils/fromEntries';

export const QCOVID_PAGES = [
  PATIENT_DETAILS_FORM,
  BODY_MASS_INDEX_FORM,
  CARDIOVASCULAR_DISEASES_FORM,
  RESP_DISEASES_TREATMENTS_FORM,
  METABOLIC_RENAL_FORM,
  NEURO_PSYCH_CONDITIONS_FORM,
  IMMU_HAEMA_CONDITIONS_FORM,
  IMMUNO_CANCER_CONDITIONS_FORM,
];

export const QCOVID_FORMS = fromEntries(
  QCOVID_PAGES.map((form) => [form.formId, form])
);

export const QCOVID_PAGE_NAMES = QCOVID_PAGES.map((form) => form.formId);

export const QCOVID_DEFAULTS = extractDefaults(QCOVID_FORMS);
