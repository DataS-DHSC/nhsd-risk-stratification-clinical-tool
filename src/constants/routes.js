export const ROUTE_HOME = '/';
export const ROUTE_LOGIN = '/login';
export const ROUTE_PATIENT_DETAILS = '/patient-details';
export const ROUTE_DEMOGRAPHIC_INFO = '/demographic';
export const ROUTE_BODY_MASS_INDEX = '/body-mass-index';
export const ROUTE_MEDICAL_INFO = '/medical-info';
export const ROUTE_METABOLIC_RENAL_LIVER = '/metabolic-renal-liver-conditions';
export const ROUTE_CARDIOVASCULAR_DISEASES = '/cardiovascular-diseases';
export const ROUTE_RESPIRATORY_DISEASES =
  '/respiratory-diseases-and-treatments';
export const ROUTE_AUTOIMMUNE_AND_HAEMATOLOGICAL =
  '/immune-and-haematological-conditions';
export const ROUTE_NEUROLOGICAL_AND_PSYCHIATRIC =
  '/neurological-and-psychiatric-conditions';
export const ROUTE_IMMUNOSUPRESSANTS_CANCER_CODNDITIONS =
  '/immunosuppressants-cancer-conditions';
export const ROUTE_CHECK_ANSWERS = '/check-your-answers';
export const ROUTE_SCORE = '/score';
export const ROUTE_QCOVID = '/ras/qcovid';
export const ROUTE_IP_ENDPOINT = '/ras/ip';

export const ROUTE_SUPPORT_FORM = 'https://www.nhs.uk/contact-us/covid-risk';

export const ROUTE_CLINICAL_GUIDANCE =
  'https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians';
export const ROUTE_PATIENT_GUIDANCE =
  'https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/information-for-patients';
export const ROUTE_ACCESS_INFO =
  'https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool#access-the-tool';

export const ROUTE_PAGE = '/page';
export const ROUTE_COOKIES_POLICY = `${ROUTE_PAGE}/cookies/policy`;
export const ROUTE_COOKIES_SETTINGS = `${ROUTE_PAGE}/cookies/settings`;
export const ROUTE_COOKIES_CONFIRMATION = `${ROUTE_PAGE}/cookies/confirmation`;
export const ROUTE_ACCESSIBILITY = `${ROUTE_PAGE}/accessibility`;
export const ROUTE_FEEDBACK = `/ras${ROUTE_PAGE}/feedback`;
export const ROUTE_TERMS_OF_USE = `${ROUTE_PAGE}/terms-of-use`;
export const ROUTE_OPEN_SOURCE = `${ROUTE_PAGE}/open-source`;
export const ROUTE_CE_MARK = `${ROUTE_PAGE}/ce-mark-label`;
export const ROUTE_SUPPORTED_BROWSERS = `${ROUTE_PAGE}/supported-browsers`;

export const ROUTE_AGE = '/age';
export const ROUTE_ETHNIC_GROUP = '/ethnic-group';
export const ROUTE_POSTCODE = '/postcode';
export const ROUTE_KIDNEY_DISEASE = '/kidney-disease';
export const ROUTE_LEARNING_DISABILITIES_OR_DOWNS_SYNDROME =
  '/learning-disability-or-downs-syndrome';
export const ROUTE_CHEMOTHERAPY = '/chemotherapy-in-the-last-12-months';
