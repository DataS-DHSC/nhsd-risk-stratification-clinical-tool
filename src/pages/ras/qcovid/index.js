import PropTypes from 'prop-types';
import HeadTags from '../../../components/HeadTags';
import Start from '../../../components/qcovid/Start';
import { ROUTE_QCOVID, ROUTE_PATIENT_DETAILS } from '../../../constants/routes';
import handleServerSideProps from '../../../utils/handleServerSideProps';

const Index = ({ prevData, featureFlags }) => {
  const d = new Date();
  d.setTime(d.getTime() + 12 * 24 * 60 * 60 * 1000);

  return (
    <>
      <HeadTags title="" description="" featureFlags={featureFlags} />
      <Start route={ROUTE_QCOVID + ROUTE_PATIENT_DETAILS} prevData={prevData} />
    </>
  );
};

Index.propTypes = {
  prevData: PropTypes.shape().isRequired,
  featureFlags: PropTypes.shape(),
};

Index.defaultProps = {
  featureFlags: {},
};

export const getServerSideProps = handleServerSideProps({ allowGet: true });

export default Index;
