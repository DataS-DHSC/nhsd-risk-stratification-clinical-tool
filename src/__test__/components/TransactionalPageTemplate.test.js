import React from 'react';
import { render } from '@testing-library/react';
import TransactionalPageTemplate from '../../components/TransactionalPageTemplate';
import { ROUTE_QCOVID } from '../../constants/routes';
import { QCOVID_PAGE_NAMES } from '../../constants/forms-metadata/qcovid/qcovid-forms';

const prevData = {
  field1: 'field1prevData',
  field2: 'field2prevData',
  csrfToken: 'csrfToken',
  headers: '{"correlationId":"5383302b-f729-4076-97d5-c428b26ef5dd"}',
};

const contextData = {
  field1: 'field1ContextData',
  field2: 'field1ContextData',
};

describe('transactional page template', () => {
  it('snapshot should be built from prevData when client side routing disabled', () => {
    const context = jest.fn();
    const form = { formId: 'form-id', label: 'form-label', sections: [] };
    context.mockImplementationOnce(() => [null, jest.fn()]);

    const container = render(
      <TransactionalPageTemplate
        prevData={prevData}
        form={form}
        context={context}
        reviewDatesSource="qcovid"
        formBaseUrl={ROUTE_QCOVID}
        pageNames={QCOVID_PAGE_NAMES}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('snapshot should be built from context data when client side routing enabled', () => {
    const context = jest.fn();
    const form = { formId: 'form-id', label: 'form-label', sections: [] };
    context.mockImplementationOnce(() => [contextData, jest.fn()]);

    const container = render(
      <TransactionalPageTemplate
        prevData={prevData}
        form={form}
        context={context}
        reviewDatesSource="qcovid"
        formBaseUrl={ROUTE_QCOVID}
        pageNames={QCOVID_PAGE_NAMES}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
