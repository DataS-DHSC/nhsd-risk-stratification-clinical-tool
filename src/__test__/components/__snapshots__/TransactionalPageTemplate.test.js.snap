// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transactional page template snapshot should be built from context data when client side routing enabled 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-two-thirds"
    >
      <h1
        class="nhsuk-fieldset__legend--l"
        id="form-label-page-heading"
      >
        form-label
      </h1>
      <form
        action="/ras/qcovid/patient-details"
        id="form-id-form"
        method="POST"
      >
        <input
          name="field1"
          type="hidden"
          value="field1ContextData"
        />
        <input
          name="field2"
          type="hidden"
          value="field1ContextData"
        />
        <input
          name="csrfToken"
          type="hidden"
          value="csrfToken"
        />
        <button
          aria-disabled="false"
          class="nhsuk-button nhsuk-u-margin-bottom-8"
          id="continue-button"
          type="submit"
        >
          Continue
        </button>
      </form>
      <div
        class="nhsuk-review-date"
        id="review-date"
      >
        <p
          class="nhsuk-body-s"
        >
          Page last reviewed: 1 December 2020
          <br />
          Next review due: 23 December 2023
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`transactional page template snapshot should be built from prevData when client side routing disabled 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-two-thirds"
    >
      <h1
        class="nhsuk-fieldset__legend--l"
        id="form-label-page-heading"
      >
        form-label
      </h1>
      <form
        action="/ras/qcovid/patient-details"
        id="form-id-form"
        method="POST"
      >
        <input
          name="field1"
          type="hidden"
          value="field1prevData"
        />
        <input
          name="field2"
          type="hidden"
          value="field2prevData"
        />
        <input
          name="csrfToken"
          type="hidden"
          value="csrfToken"
        />
        <input
          name="headers"
          type="hidden"
          value="{\\"correlationId\\":\\"5383302b-f729-4076-97d5-c428b26ef5dd\\"}"
        />
        <button
          aria-disabled="false"
          class="nhsuk-button nhsuk-u-margin-bottom-8"
          id="continue-button"
          type="submit"
        >
          Continue
        </button>
      </form>
      <div
        class="nhsuk-review-date"
        id="review-date"
      >
        <p
          class="nhsuk-body-s"
        >
          Page last reviewed: 1 December 2020
          <br />
          Next review due: 23 December 2023
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
